src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs

```
@@ -203,42 +187,11 @@ public class ReverseEngineeringGenerator
             }
         }
 
-        public virtual string GetDbContextTemplateFileName([NotNull] IDatabaseMetadataModelProvider provider)
-        {
-            Check.NotNull(provider, nameof(provider));
-
-            return provider.GetType().GetTypeInfo().Assembly.GetName().Name + "." + DbContextTemplateFileName;
-        }
-
-        public virtual string GetEntityTypeTemplateFileName([NotNull] IDatabaseMetadataModelProvider provider)
-        {
-            Check.NotNull(provider, nameof(provider));
+        public virtual string GetDbContextTemplateFileName() 
+            => _provider.GetType().GetTypeInfo().Assembly.GetName().Name + "." + DbContextTemplateFileName;
 
-            return provider.GetType().GetTypeInfo().Assembly.GetName().Name + "." + EntityTypeTemplateFileName;
-        }
-
-        private static void CheckConfiguration(ReverseEngineeringConfiguration configuration)
-        {
```

Am I missing where you are checking these now? In my other PR for instance this is where we check that the output path passed in is relative. I'd prefer to fail fast with a helpful message.